src/client/devapi_base.cpp

```diff
@@ -4480,234 +5146,271 @@ namespace Tango {
             }
-            catch (CORBA::COMM_FAILURE &comm) {
-                if (comm.minor() == omni::COMM_FAILURE_WaitingForReply) {
-                    TRANSIENT_NOT_EXIST_EXCEPT(comm, "DeviceProxy", "write_pipe", this);
-                } else {
-                    dev_pipe.get_root_blob().reset_insert_ctr();
-                    delete tmp_ptr;
-
-                    set_connection_state(CONNECTION_NOTOK);
-                    stringstream desc;
-                    desc << "Failed to write_pipe on device " << device_name;
-                    ApiCommExcept::re_throw_exception(comm, "API_CommunicationFailed", desc.str(),
-                                                      "DeviceProxy::write_pipe()");
-                }
+        }
+        catch (CORBA::COMM_FAILURE &comm)
+        {
+            if (comm.minor() == omni::COMM_FAILURE_WaitingForReply)
+            {
+                TRANSIENT_NOT_EXIST_EXCEPT(comm, "DeviceProxy", "write_pipe", this);
             }
-            catch (CORBA::SystemException &ce) {
+            else
+            {
                 dev_pipe.get_root_blob().reset_insert_ctr();
-                delete tmp_ptr;
 
                 set_connection_state(CONNECTION_NOTOK);
                 stringstream desc;
                 desc << "Failed to write_pipe on device " << device_name;
-                ApiCommExcept::re_throw_exception(ce, "API_CommunicationFailed", desc.str(),
+                ApiCommExcept::re_throw_exception(comm, "API_CommunicationFailed", desc.str(),
                                                   "DeviceProxy::write_pipe()");
             }
         }
+        catch (CORBA::SystemException &ce)
+        {
+            dev_pipe.get_root_blob().reset_insert_ctr();
+
+            set_connection_state(CONNECTION_NOTOK);
+            stringstream desc;
+            desc << "Failed to write_pipe on device " << device_name;
+            ApiCommExcept::re_throw_exception(ce, "API_CommunicationFailed", desc.str(),
+                                              "DeviceProxy::write_pipe()");
+        }
+    }
+
+    dev_pipe.get_root_blob().reset_insert_ctr();
+
+}
+
+DevPipeData
+DeviceProxy::createDevPipeData(DevicePipe &dev_pipe) const
+{
+    DevPipeData result;
 
-        dev_pipe.get_root_blob().reset_insert_ctr();
-        delete tmp_ptr;
+    result.name = dev_pipe.get_name().c_str();
+    const string &bl_name = dev_pipe.get_root_blob().get_name();
+    if (bl_name.size() != 0)
+        result.data_blob.name = bl_name.c_str();
+
+
+    DevVarPipeDataEltArray *tmp_ptr = dev_pipe.get_root_blob().get_insert_data();
+    if (tmp_ptr == Tango_nullptr)
+    {
+        Except::throw_exception(API_PipeNoDataElement, "No data in pipe!", "DeviceProxy::write_pipe()");
     }
 
+    DevULong max, len;
+    max = tmp_ptr->maximum();
+    len = tmp_ptr->length();
+    result.data_blob.blob_data.replace(max, len, tmp_ptr->get_buffer((DevBoolean) true), true);
+    delete tmp_ptr;
```

Comment:
Are you sure it is correct to delete tmp_ptr before `dev->write_pipe_5(pipe_value_5, ci);` statement?
Please be extra careful with the meaning of 2 boolean parameters passed in:
`result.data_blob.blob_data.replace(max, len, tmp_ptr->get_buffer((DevBoolean) true), true);` statement because there are some memory ownership transfer involved here.